processing/src/main/java/org/apache/druid/query/context/ResponseContext.java

```diff
@@ -112,6 +114,30 @@
         "uncoveredIntervalsOverflowed",
             (oldValue, newValue) -> (boolean) oldValue || (boolean) newValue
     ),
+    /**
+     * Map of most relevant query ID to remaining number of responses from query nodes.
+     * The value is initialized in {@code CachingClusteredClient} when it initializes the connection to the query nodes,
+     * and is updated whenever they respond (@code DirectDruidClient). {@code RetryQueryRunner} uses this value to
+     * check if the {@link #MISSING_SEGMENTS} is valid.
+     *
+     * Currently, the broker doesn't run subqueries in parallel, the remaining number of responses will be updated
+     * one by one per subquery. However, since we are planning to parallelize running subqueries, we store them
```

Comment:
> I'm not sure if we are, but, we also haven't decided definitely _not_ to.

Heh, rephrased to sound more chill.

> I think it would also be okay to not handle parallel queries here, but instead build in a sanity check that verifies the subqueries are issued in series. Maybe by verifying that when the id changes, the previous number of responses remaining must be down to zero. It's up to you. I think your current code is okay too.

Hmm, how could it be used? Is it to make sure that the subqueries will be issued in series until we parallelize running them?